src/main/java/com/epam/page/object/generator/containers/SupportedTypesContainer.java

```diff
@@ -42,37 +43,59 @@ public SupportedTypesContainer() {
     }
 
     private void fillSupportedTypesMap() {
-        supportedTypesMap.put("button", new ClassAndAnnotationPair(Button.class, FindBy.class));
-        supportedTypesMap.put("text", new ClassAndAnnotationPair(Text.class, FindBy.class));
-        supportedTypesMap.put("checkbox", new ClassAndAnnotationPair(CheckBox.class, FindBy.class));
-        supportedTypesMap.put("image", new ClassAndAnnotationPair(Image.class, FindBy.class));
+        supportedTypesMap.put(SearchRuleType.BUTTON.getName(),
```

Comment:
I think it's better to use SearchType instead of String in supportedTypesMap as we introduces a separate class for representing a search type.

Reply:
Yes, but it will be difficult to compare SearchRuleType with String from SearchRule.getType()